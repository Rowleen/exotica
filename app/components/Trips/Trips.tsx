import { FC } from 'react'
import type { Trip } from '../../core/domain/entities/Trip'
import TripCard from '../TripCard/TripCard'

import { modalName } from '../../shared/types/modalName'

import styles from './trips.module.sass'

interface TripsProps {
  trips: Trip[] | undefined
  handleSelectTrip: (tripId: number, modalName: modalName) => void
  handleDeleteTrip: (tripId: number) => void
}

const Trips: FC<TripsProps> = ({
  trips,
  handleSelectTrip,
  handleDeleteTrip
}) => {
  return (
    <section className={styles.trips}>
      {trips ? (
        trips.map((trip, index) => (
          <TripCard
            key={index + trip.id}
            tripId={trip.id}
            title={trip.title}
            photoUrl={trip.photo_url}
            description={trip.description}
            handleSelectTrip={handleSelectTrip}
            handleDeleteTrip={handleDeleteTrip}
          />
        ))
      ) : (
        <h2>We&apos;re so sorry. We haven&apos;t trips yet ðŸ˜“</h2>
      )}
    </section>
  )
}

export default Trips
