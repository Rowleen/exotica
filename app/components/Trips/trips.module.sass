@import '../../sass/variables'

.trips
  width: 100%
  display: grid
  grid-template-rows: 1fr
  row-gap: $spacing-3xl