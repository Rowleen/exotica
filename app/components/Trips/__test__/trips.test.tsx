import '@testing-library/jest-dom'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render, screen } from '@testing-library/react'
import Trips from '../Trips'
import trips from '../../../__mocks__/trips.mock.json'

const queryClient = new QueryClient()

describe('Trips component', () => {
  const renderComponent = (
    <QueryClientProvider client={queryClient}>
      <Trips
        trips={trips}
        handleSelectTrip={jest.fn()}
        handleDeleteTrip={jest.fn()}
      />
    </QueryClientProvider>
  )

  it('renders Trips unchanged', () => {
    const { container } = render(renderComponent)
    expect(container).toMatchSnapshot()
  })

  it('render trip', () => {
    const { getByText } = render(renderComponent)

    const h2 = getByText('Hawaii')
    expect(h2).toBeInTheDocument()
  })

  it('Does not show trip wich it is not have been created', () => {
    const { queryByText } = render(renderComponent)

    const h2 = queryByText('EEUU')
    expect(h2).not.toBeInTheDocument()
  })

  it('render all trips', () => {
    const { getAllByRole } = render(renderComponent)

    const headings = getAllByRole('heading')
    expect(headings).toHaveLength(7)
  })
})
