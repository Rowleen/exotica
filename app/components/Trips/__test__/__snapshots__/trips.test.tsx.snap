// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trips component renders Trips unchanged 1`] = `
<div>
  <section
    class="trips"
  >
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://a.cdn-hotels.com/gdcs/production82/d1923/447a348f-f875-4885-b00a-e9a90603fef5.jpg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            Portugal
          </h2>
          <p>
            Embark on a journey through Portugal, where the charming streets of Lisbon captivate you, the golden beaches of the Algarve await, and Portuguese cuisine delights with authentic flavors. Explore castles in Sintra and create unforgettable memories in this destination full of history and beauty. Portugal invites you to experience something truly unique!
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://cdn.aarp.net/content/dam/aarp/travel/Domestic/2021/12/1140-oahu-hero.jpg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            Hawaii
          </h2>
          <p>
            Embark on a tropical escapade to Hawaii, where the lush landscapes of Oahu, the volcanic wonders of Maui, and the serene beaches of Kauai beckon. Immerse yourself in the Aloha spirit, savor the flavors of traditional Hawaiian cuisine, and witness the breathtaking beauty of the Pacific archipelago. Hawaii invites you to experience the perfect blend of paradise and adventure.
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://content.skyscnr.com/m/6a82667a63ac12a3/original/GettyImages-156384414.jpg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            China
          </h2>
          <p>
            Explore China, from the bustling streets of Beijing to the ancient wonders of the Great Wall. Delight your senses with exquisite Chinese cuisine, explore temples in Xi'an, immerse yourself in the modernity of Shanghai, and create unforgettable memories in this destination full of history and beauty. China invites you to experience a unique journey, where rich tradition meets 21st-century innovation.
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://humanidades.com/wp-content/uploads/2017/07/japon-7-e1571188423405-800x415.jpg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            Japan
          </h2>
          <p>
            Embark on a captivating journey to Japan, where the seamless blend of ancient traditions and modern innovation awaits. Immerse yourself in Tokyo's vibrant streets, explore historic temples in Kyoto, and savor the delicate flavors of authentic Japanese cuisine. From the iconic Mount Fuji to the serene gardens of Kanazawa, Japan invites you to discover a captivating blend of cultural richness and natural beauty.
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://cdn.britannica.com/60/92960-050-327CF926/pyramid-Mayan-Chichen-Itza-Mex.jpg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            Mexico
          </h2>
          <p>
            Embark on an immersive journey through Mexico, where the vibrant culture of Mexico City beckons, the ancient wonders of Chichen Itza await, and the spicy delights of Mexican cuisine tantalize your taste buds. From the historic charm of Oaxaca to the pristine beaches of Tulum, Mexico invites you to delve into a rich tapestry of experiences that showcase its diverse landscapes and cultural treasures.
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://lp-cms-production.imgix.net/2019-06/09a64fea2933f6da77ab07d671d1f678-south-korea.jpg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            South Korea
          </h2>
          <p>
            Explore the enchanting landscapes and rich cultural heritage of South Korea. From the bustling streets of Seoul to the historic sites of Gyeongju, and the serene beauty of Jeju Island, South Korea invites you on an immersive journey through its vibrant cities and picturesque landscapes.
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
    <article
      class="card"
    >
      <div
        class="photo"
        style="background-image: url(https://miro.medium.com/v2/resize:fit:1400/1*h_nQx6iZZ3wUe-RTgzlMjw.jpeg); background-size: cover;"
      />
      <div
        class="details"
      >
        <div
          class="description"
        >
          <h2
            class="title"
          >
            Slovenia
          </h2>
          <p>
            Embark on a picturesque journey to Slovenia, where the charming streets of Ljubljana enchant, the emerald waters of Lake Bled captivate, and the flavors of Slovenian cuisine delight your taste buds. From the medieval charm of Å kofja Loka to the majestic Triglav National Park, Slovenia invites you to explore its diverse landscapes and embrace the warmth of its rich cultural heritage.
          </p>
        </div>
        <div
          class="actionButtons"
        >
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              See trip details
            </button>
          </div>
          <div
            class="column"
          >
            <button
              class="link"
              type="button"
            >
              Edit
            </button>
            <button
              class="link linkDanger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </article>
  </section>
</div>
`;
